templates/common/_base/units/ovs-configuration.service.yaml

```diff
@@ -8,8 +8,8 @@ contents: |
   # This service is used to move a physical NIC into OVS and reconfigure OVS to use the host IP
   Requires=openvswitch.service
   Wants=NetworkManager-wait-online.service
-  After=NetworkManager-wait-online.service openvswitch.service node-valid-hostname.service
-  Before=network-online.target kubelet.service crio.service
+  After=NetworkManager-wait-online.service openvswitch.service
+  Before=network-online.target kubelet.service crio.service node-valid-hostname.service
```

Comment:
since ovs-configuration is part of network-online.target, and really node-valid-hostname needs DHCP to be done (regardless of if ovs-configuration is present or not), it seems like it would be better in the node-valid-hostname.service file to do:

After=network-online.target
Wants=network-online.target

Reply:
That reasoning makes sense to me, but we'd also need to change the kubelet service then because it currently waits for network-online too. To avoid reintroducing the race we'd need to have kubelet explicitly depend on node-valid-hostname.

I went back and looked at the original PR for node-valid-hostname, and using network-online was the mechanism chosen to block other services from starting if it failed. I don't know if there was a reason we didn't just add a dependency on node-valid-hostname directly. Maybe @darkmuggle can chime in.

Reply:
> I don't know if there was a reason we didn't just add a dependency on node-valid-hostname directly.  

The reason I proposed using `network-online.target` was to provide a blocking mechanism for all the other things that changing the hostname breaks -- logging, cri-o, kubelet, ssh, etc....  The goal was to put to bed all the `localhost` problems by blocking signaling that networking is not really up until we have a valid hostname. In evaluating the problem, `network-online.target` was the lowest blast radius and it didn't require changing all the up-stream services.  The other big reason was for debugging: we block things from coming up before the hostname is valid and that makes it really clear why the boot failed. The pantheon of bugs related to the hostname being cluster-invalid was a bit dizzying.